core/src/main/scala/caliban/validation/Validator.scala

```diff
@@ -823,27 +860,49 @@ object Validator {
   private[caliban] def validateInputObject(t: __Type): EReader[Any, ValidationError, Unit] = {
     val inputObjectContext = s"""InputObject '${t.name.getOrElse("")}'"""
 
-    def noDuplicateInputValueName(
-      inputValues: List[__InputValue],
-      errorContext: String
-    ): EReader[Any, ValidationError, Unit] = {
-      val messageBuilder = (i: __InputValue) => s"$errorContext has repeated fields: ${i.name}"
+    def noDuplicateInputValueName(inputValues: List[__InputValue]): EReader[Any, ValidationError, Unit] = {
+      val messageBuilder = (i: __InputValue) => s"$inputObjectContext has repeated fields: ${i.name}"
       val explanatory    =
         "The input field must have a unique name within that Input Object type; no two input fields may share the same name"
       noDuplicateName[__InputValue](inputValues, _.name, messageBuilder, explanatory)
     }
 
+    def noDuplicatedOneOfOrigin(inputValues: List[__InputValue]): EReader[Any, ValidationError, Unit] = {
+      val resolveOrigin  = (i: __InputValue) => i.parentTypeName.getOrElse("<unexpected validation error>")
+      val messageBuilder = (i: __InputValue) =>
+        s"OneOf $inputObjectContext has multiple arguments from the same case class: ${resolveOrigin(i)}"
+      val explanatory    = "All case classes used as arguments to OneOf Input Objects must have exactly one field"
+      noDuplicateName[__InputValue](inputValues, resolveOrigin, messageBuilder, explanatory)
+    }
+
     def validateFields(fields: List[__InputValue]): EReader[Any, ValidationError, Unit] =
       ZPure.foreachDiscard(fields)(validateInputValue(_, inputObjectContext)) *>
-        noDuplicateInputValueName(fields, inputObjectContext)
+        noDuplicateInputValueName(fields)
+
+    def validateOneOfFields(fields: List[__InputValue]): EReader[Any, ValidationError, Unit] =
+      failValidationWhen(fields.size <= 1)(
+        s"OneOf $inputObjectContext has less than 2 fields",
+        "OneOf input objects must have at least 2 possible fields"
+      ) *>
```

Comment:
This is not mentioned anywhere in the spec (i.e., it's not explicitly allowed or disallowed), but it doesn't seem to me like something that should be allowed normally.

The only case I can think of that might require this is if you currently have only 1 type of input but you plan to add more cases in the future and don't want to introduce a breaking change..

Thoughts?